be/btree.c

```diff
@@ -505,18 +520,22 @@ static void be_btree_split_child(struct m0_be_btree *btree,
 	for (i = parent->bt_num_active_key + 1; i > index + 1; i--) {
 		parent->bt_child_arr[i] = parent->bt_child_arr[i - 1];
 		parent->bt_kv_arr[i - 1] = parent->bt_kv_arr[i - 2];
+		parent->bt_ik[i - 1] = parent->bt_ik[i - 2];
+		parent->allocated[i - 1] = parent->allocated[i - 2];
 		if(btree->bb_ops->ko_type != M0_BBT_CAS_CTG)
 		{
-			parent->bt_kv_arr[i - 1].btree_key = &parent->bt_kv_arr[i - 1].inlkey;
+			parent->bt_kv_arr[i - 1].btree_key = &parent->bt_ik[i - 1].inlkey;
```

Comment:
Please go through this complete function and see if the new changes make sense or they need rework.